test/boost/repair.cc

```diff
@@ -0,0 +1,137 @@
+#include "gc_clock.hh"
+#include "sstables/sstable_writer.hh"
+#include "readers/upgrading_consumer.hh"
+#include "readers/from_fragments_v2.hh"
+#include "test/lib/random_utils.hh"
+
+class test_mutation_fragment_queue_impl : public mutation_fragment_queue::impl, enable_lw_shared_from_this<test_mutation_fragment_queue_impl>{
+    schema_ptr _schema;
+    reader_permit _permit;
+    std::deque<mutation_fragment_v2> _fragments;
+    std::unique_ptr<upgrading_consumer<std::function<void(mutation_fragment_v2&&)>>> _consumer;
+    shared_sstable _sst;
+    sstable_writer_config _cfg;
+public:
+    test_mutation_fragment_queue_impl(schema_ptr s, reader_permit permit
+            , shared_sstable sst
+            , sstable_writer_config cfg)
+        : mutation_fragment_queue::impl()
+        , _schema(std::move(s))
+        , _permit(std::move(permit))
+        , _sst(std::move(sst))
+        , _cfg(std::move(cfg))
+    {
+        _consumer = std::make_unique<upgrading_consumer<std::function<void(mutation_fragment_v2&&)>>>
+            (*_schema, _permit, [this](mutation_fragment_v2&& mf) mutable {
+                _fragments.push_back(std::move(mf));
+            });
+    }
+    
+    virtual future<> push(mutation_fragment mf) override {
+        _consumer->consume(std::move(mf));
+        return make_ready_future();
+    }
+
+    virtual void abort(std::exception_ptr ep) override {}
+
+    virtual void push_end_of_stream() override {
+        flat_mutation_reader_v2 r = make_flat_mutation_reader_from_fragments(_schema, _permit, std::move(_fragments));
+        int estimated_partitions = 1;
+        _sst->write_components(std::move(r), estimated_partitions, _schema, _cfg, encoding_stats{}).get();
+        _sst->load().get();
+    }
+};
+
+mutation_fragment_queue make_test_mutation_fragment_queue(schema_ptr s, reader_permit permit, shared_sstable sst, sstable_writer_config cfg) {
+    return mutation_fragment_queue(std::make_unique<test_mutation_fragment_queue_impl>(std::move(s), std::move(permit), std::move(sst), std::move(cfg)));
+}
+
+class test_repair_writer_impl : public repair_writer::impl {
+    mutation_fragment_queue _queue;
+public:
+    test_repair_writer_impl(mutation_fragment_queue queue)
+        : _queue(std::move(queue))
+    {
+    }
+
+    virtual future<> wait_for_writer_done() override {
+        // Pushing EOS to the underlying mq waits until the sstable is written
+        return make_ready_future();
+    }
+
+    virtual mutation_fragment_queue& queue() override {
+        return _queue;
+    }
+
+    virtual void start_streaming(lw_shared_ptr<repair_writer> writer) override {}
+};
+
+lw_shared_ptr<repair_writer> make_test_repair_writer(schema_ptr schema, reader_permit permit, mutation_fragment_queue mq) {
+    return make_lw_shared<repair_writer>(std::move(schema), std::move(permit), std::make_unique<test_repair_writer_impl>(std::move(mq)));
+}
+
+SEASTAR_TEST_CASE(flush_repair_rows_on_wire_to_sstable) {
```

Comment:
If this test is a unit-test for a specific class or function, please add a comment with a concrete explanation of what function or functions this test case checks. There's a lot (over 130 lines) of code here and it's hard to understand what is being tested and what is just an implementation detail of the test.

Reply:
Reorganised the code and added comments